### PR TITLE
Change nodes size/positioning from pixels to slots

### LINKED ISSUE #1679: Change nodes size/positioning from pixels to slots
Currently, xodballs express position and size values in pixels. This breaks the layout of all patches created before a grid size change, such as #1678 

It is a good idea to keep the values in units which are relative to the slot size regardless of the slot absolute size so that `{x: 34, y: 102}` becomes `{x: "1s", y: "1s"}` where the `"s"` denotes “slot” units. The slot units might have fractions to express unsnapped nodes.

### Acceptance criteria

- [x] On loading a project, all dimensional values stored in pixels are converted to slot units by using the geometry coefficients of XOD 0.27 grid (x = 34px per slot, y = 102px per slot)
- [x] The slot values are converted to pixels using the current grid geometry
- [x] When a project is saved, all dimensional values are converted to slot units and stored this way
- [x] When pixels are converted to slots they are rounded to the nearest integer if fall within ±5%
- [x] As an end-to-end consequence, if a patch created in 0.27 and below has all its nodes grid-aligned, they are still grid aligned after the #1678 grid change

### PR DESCRIPTION
**Pull request by brusherru**
It closes #1679 .

### CODE REVIEW

File: packages/xod-project/src/comment.js [hunk 1]
```diff
@@ -39,7 +49,8 @@ export const getCommentSize = def(
 
 export const setCommentSize = def(
   'setCommentSize :: Size -> Comment -> Comment',
-  R.assoc('size')
+  (size, comment) =>
+    R.compose(R.assoc('size', R.__, comment), normalizeSize)(size)
```
Comment (nkrkv) [inline packages/xod-project/src/comment.js]: Can be simpler:

```js
R.assoc('size', normalizeSize(size), comment)
```

File: packages/xod-project/src/comment.js [hunk 2]
```diff
@@ -29,7 +38,8 @@ export const getCommentPosition = def(
 
 export const setCommentPosition = def(
   'setCommentPosition :: Position -> Comment -> Comment',
-  R.assoc('position')
+  (position, comment) =>
+    R.compose(R.assoc('position', R.__, comment), normalizePosition)(position)
```
Comment (nkrkv) [inline packages/xod-project/src/comment.js]: Can be simpler (see the next comment)

File: packages/xod-project/src/expandVariadicNodes.js [hunk 3]
```diff
@@ -47,7 +48,7 @@ const createAdditionalValueTerminalGroups = (
     R.prop('value')
   )(variadicPins);
 
-  const DISTANCE_BETWEEN_TERMINALS = 50;
+  const DISTANCE_BETWEEN_TERMINALS = SLOT_SIZE.WIDTH;
```
Comment (nkrkv) [inline packages/xod-project/src/expandVariadicNodes.js]: :open_mouth: :+1:

File: packages/xod-project/src/internal/dimensionConverters.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+//
+// Dimension normalizers
+//
+//-----------------------------------------------------------------------------
+export const normalizePosition = def(
+  'normalizePosition :: Position -> Position',
+  R.ifElse(isSlotUnits, convertPositionToPixels, R.assoc('units', UNITS.PIXELS))
+);
+
+export const normalizeSize = def(
```
Comment (nkrkv) [inline packages/xod-project/src/internal/dimensionConverters.js]: A little bore. The “normalization” term in the math/arithmetics domain is derived from the word “normal”: a perpendicular to a plane or a unit-length vector.

These function names would arguably be more clear if they tell what they do. Something like `positionInPixels`/`sizeInPixels` or `toPixelSize`/`toPixelPosition`

File: packages/xod-project/src/index.js [hunk 5]
```diff
@@ -130,8 +132,12 @@ export {
   ensureLiteral,
   migrateBoundValuesToBoundLiterals,
 } from './migrations/boundValuesToBoundLiterals';
+export {
+  default as migrateOldDimensionsToSlots,
+} from './migrations/oldDimensionsToSlots';
```
Comment (nkrkv) [inline packages/xod-project/src/index.js]: I recommend avoiding words like “old” and “new” in migrations. They are too biased. To be more precise, you might use `unitlessDimensionsToSlots` or just `unitlessToSlots`

File: packages/xod-client-electron/src/app/workspaceActions.js [hunk 6]
```diff
@@ -42,6 +42,7 @@ const emitSelectProject = R.curry((send, projectMeta) =>
 // pub through rendererWindow.WebContents.send(...)
 // :: (a -> ()) -> Error -> Promise.Rejected Error
 const handleError = R.curry((send, err) => {
+  console.log(err); // eslint-disable-line no-console
```
Comment (nkrkv) [inline packages/xod-client-electron/src/app/workspaceActions.js]: :+1:

**Comment by brusherru** [comment]
@nkrkv refactored it. Check it out, please

File: packages/xod-client/src/project/nodeLayout.js [hunk 7]
```diff
@@ -47,25 +47,53 @@ export const PIN_OFFSET_FROM_NODE_EDGE = 0;
...
+// =============================================================================
+//
+// Position / Size converters
+//
+// =============================================================================
+
+// TODO: Replace with smarter one
+// :: Size -> Size
+export const pixelSizeToSlots = R.evolve({
+  width: px => Math.ceil(px / SLOT_SIZE.WIDTH),
```
Comment (nkrkv) [inline packages/xod-client/src/project/nodeLayout.js]: What’s wrong with this TODO? Looks perfectly OK to me.

File: packages/xod-client/src/project/nodeLayout.js [hunk 8]
```diff
@@ -47,25 +47,53 @@ export const PIN_OFFSET_FROM_NODE_EDGE = 0;
...
- * converts size in slots to size in pixels
- */
 // :: Size -> Size
-export const nodeSizeInSlotsToPixels = R.evolve({
+export const slotSizeToPixels = R.evolve({
   width: slots => slots * SLOT_SIZE.WIDTH,
   height: slots => R.dec(slots) * SLOT_SIZE.HEIGHT + NODE_HEIGHT,
 });
 
+// TODO: Replace with smarter one
```
Comment (nkrkv) [inline packages/xod-client/src/project/nodeLayout.js]: Ditto

**Pull request merged**